app/src/main/java/org/jellyfin/androidtv/ui/livetv/LiveTvGuideFragment.java

```diff
@@ -328,6 +330,13 @@ private boolean onKeyUp(int keyCode, KeyEvent event) {
                             showProgramOptions();
                         return true;
                     }
+                    else if (mSelectedProgramView instanceof GuideChannelHeader) {
+                        // Tuning directly to a channel
+                        GuideChannelHeader channelHeader = (GuideChannelHeader) mSelectedProgramView;
+                        BaseItemDto pChannel = channelHeader.getChannel();
+                        Lazy<ItemLauncher> itemLauncher = inject(ItemLauncher.class);
+                        itemLauncher.getValue().launch(new BaseItemDtoBaseRowItem(pChannel),null,requireContext());
```

Comment:
This has the same problem as before, the item launcher will refetch the item which is not necessary. You can just call the getItemsToPlay on playbackHelper function directly just like what happens within the item launcher.

Reply:
ah, i didnt notice itemLauncher doing the same fetching. I think we're on the same page now.